yt/visualization/plot_container.py

```diff
@@ -906,28 +906,43 @@ def set_zlim(self, field, zmin, zmax, dynamic_range=None):
 
     @invalidate_plot
     def set_cbar_minorticks(self, field, state):
-        """turn colorbar minor ticks on or off in the current plot
+        """Deprecated alias, kept for backward compatibility.
 
-        Displaying minor ticks reduces performance; turn them off
-        using set_cbar_minorticks('all', 'off') if drawing speed is a problem.
+        turn colorbar minor ticks "on" or "off" in the current plot, according to *state*
 
         Parameters
         ----------
         field : string
             the field to remove colorbar minorticks
         state : string
             the state indicating 'on' or 'off'
+        """
+        from yt.funcs import mylog
+        mylog.warning("Deprecated alias, use set_colorbar_minorticks instead.")
+
+        boolstate = {"on": True, "off": False}[state.lower()]
+        return self.set_colorbar_minorticks(field, boolstate)
 
+    @invalidate_plot
+    def set_colorbar_minorticks(self, field, state):
+        """turn colorbar minor ticks on or off in the current plot
+
+        Displaying minor ticks reduces performance; turn them off
+        using set_colorbar_minorticks('all', False) if drawing speed is a problem.
+
+        Parameters
+        ----------
+        field : string
+            the field to remove colorbar minorticks
+        state : bool
```

Comment:
Is there a reason you changed the API here from a string to a bool? I suspect a new user to yt would probably find it more intuitive to write 'on' or 'off' as an arg to set_colorbar_minorticks().

Reply:
I think it would be ok to use the bool here as long as `set_minorticks` is also updated to be consistent.

Reply:
I believe that most yt newcomers are at least familiar with matplotlib, which has a lot of such `set_<stuff>` methods and function arguments using plain boolean to decide whether a specific plot element should be visible or not, so I think it's a good opportunity to change the api to this behaviour.
I don't know which one of us has the most realistic idea of what feels intuitive to a new user, though I agree that the change should be consistently deployed to other methods.

Reply:
> I don't know which one of us has the most realistic idea of what feels intuitive to a new user

I don't really know how to feel about this, but I think both of us are trying to support our users in good faith.

Reply:
OK, so I have to say, this is a *fascinating* discussion, especially when we look at the history of `PlotWindow` and the like.  In some ways, what we've done is provide a big wrapper around a ton of matplotlib functionality to support a use case that I *think* is reasonably not-extensive, which is the idea that you have multiple fields in a single `PlotWindow` and you apply the same annotations, callbacks, etc, to all of them.  But I think for the most part, people don't do that!

So we're in a weird place where so much of what we've done has been to support a use case that we don't really focus on much in the docs, even, and that includes a fair bit of wrapping of functionality.  This isn't *bad*, and it does provide some nice ways to set defaults, but it's a layer between us and matplotlib.

Anyway, I think we should evaluate this, but I do think that any *major* API changes need to wait for 4.0, which shouldn't actually be too much longer to be `master`.

Reply:
To add to this, I personally have mixed feelings about this use case. At the same time, it makes the simple, most repeated task extremely easy to perform, which is precisely what drove me into yt in the first place, but it also makes more refined plots somewhat harder to produce when one means to "talk to matplotlib" trough yt.

Anyway, I'm absolutely fine with delaying these changes to yt-4.0 if you think it's appropriate. However, I'm not sure I'm reading your statement right. Do you consider this a _major_ change ? I guess not, but I'll admit I'm unsure.

Reply:
I suspect these changes are not major API changes and can be merged before yt-4.0 (though Matt can also weigh in here if I've misinterpreted things). I think this is especially true because you've kept the old functions and have some depreciation warnings when they're called. You've certainly added to the usability of yt and I think this is a helpful contribution. 🙂

Reply:
I agree!